mlflow/models/evaluation/base.py

```diff
@@ -778,6 +798,70 @@ def evaluate(
+                                the model. The ``target`` column contains the corresponding labels
+                                to the predictions made on that row.
+                              - Dict: The second is a dictionary containing the metrics calculated
+                                by the default evaluator. The keys are the names of the metrics
+                                and the values are the scalar values of the metrics. Refer to the
+                                DefaultEvaluator behavior section for what metrics will be returned
+                                based on the type of model (i.e. classifier or regressor).
+
+                              A custom metric function can return in two different formats:
+                              - Dict[AnyStr, Union[int, float, np.number]: a singular dictionary of
+                                custom metrics, where the keys are the names of the metrics, and the
+                                values are the scalar values of the metrics.
+                              - Tuple[Dict[AnyStr, Union[int,float,np.number]], Dict[AnyStr,Any]]:
+                                a tuple of a dict containing the custom metrics, and a dict of
+                                artifacts, where the keys are the names of the artifacts, and the
+                                values are objects representing the artifacts.
+
+                              Object types that artifacts be represented as:
+                              - A string uri representing the file path to the artifact. MLflow will
+                                infer the type of the artifact based on the file extension.
+                              - ``EvaluationArtifact`` type objects. i.e. CsvEvaluationArtifact,
+                                ImageEvaluationArtifact.
+                              - Pandas DataFrame. This will be resolved as a CSV artifact.
+                              - Numpy array. This will be saved as a .npy artifact.
+                              - Matplotlib Figure. This will be saved as an image artifact.
+                              - Any JSON serializable object. This will be saved as a JSON artifact.
+
+                              Custom Metric Function Boilerplate:
+                                  ``
+                                  def custom_metrics_boilerplate(eval_df, builtin_metrics):
+                                      ...
+                                      metrics: Dict[AnyStr, Union[int, float, np.number]] = ???
+                                      artifacts: Dict[AnyStr, Any] = ???
+                                      ...
+                                      if artifacts is not None:
+                                          return metrics, artifacts
+                                      return metrics
+                                  ``
+
+                              Example Usage:
```

Comment:
Can we use `.. code-block: python`, as in https://github.com/mlflow/mlflow/blob/2c0d13db6c677f1023a9a2a1776aca9e32aebe3d/mlflow/tracking/fluent.py#L208-L235?